app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt

```diff
@@ -261,28 +272,78 @@ class TabPreferenceActivity : BaseActivity(), Injectable, ItemInteractionListene
+        progress.setPadding(preferredPadding, 0, preferredPadding, 0)
+        progress.visible(false)
+
+        val dialogBuilder = AlertDialog.Builder(this)
+            .setTitle(R.string.select_list_title)
+            .setNegativeButton(android.R.string.cancel, null)
+            .setView(progress)
+            .setAdapter(adapter) { _, position ->
+                val list = adapter.getItem(position)
+                val newTab = createTabDataFromId(LIST, listOf(list!!.id, list.title))
+                currentTabs.add(newTab)
+                currentTabsAdapter.notifyItemInserted(currentTabs.size - 1)
+                updateAvailableTabs()
+                saveTabs()
+            }
+
+        val loadListProgressBarJob = getProgressBarJob(progress, 500)
+
+        if (adapter.isEmpty) {
+            dialogBuilder.setTitle(R.string.select_list_title_empty)
+            dialogBuilder.setNeutralButton(R.string.select_list_create) { _, _ ->
+                val listIntent = Intent(applicationContext, ListsActivity::class.java)
+                startActivity(listIntent)
+            }
+
+            loadListProgressBarJob.start()
+        }
+
+        val dialog = dialogBuilder.show()
+
         lifecycleScope.launch {
             mastodonApi.getLists().fold(
                 { lists ->
+                    loadListProgressBarJob.cancel()
                     adapter.addAll(lists)
+                    if (lists.isNotEmpty()) {
+                        dialog.getButton(DialogInterface.BUTTON_NEUTRAL).hide()
+                        dialog.setTitle(R.string.select_list_title)
+                    }
                 },
                 { throwable ->
+                    dialog.hide()
                     Log.e("TabPreferenceActivity", "failed to load lists", throwable)
+                    Snackbar.make(binding.root, R.string.error_list_load, Snackbar.LENGTH_LONG).show()
                 }
             )
         }
+    }
 
-        AlertDialog.Builder(this)
-            .setTitle(R.string.select_list_title)
-            .setAdapter(adapter) { _, position ->
-                val list = adapter.getItem(position)
-                val newTab = createTabDataFromId(LIST, listOf(list!!.id, list.title))
-                currentTabs.add(newTab)
-                currentTabsAdapter.notifyItemInserted(currentTabs.size - 1)
-                updateAvailableTabs()
-                saveTabs()
-            }
-            .show()
+    private fun getDimensionFromAttribute(context: Context, attr: Int): Int {
+        val typedValue = TypedValue()
+        return if (context.theme.resolveAttribute(attr, typedValue, true))
+            TypedValue.complexToDimensionPixelSize(typedValue.data, context.resources.displayMetrics)
+        else 0
+    }
+
+    // TODO this should be made general somewhere? (it is copied from ViewThreadFragment)
+    //   also: maybe it should have a callable instead of simply view.show()?
```

Comment:
Tried this now (two functions as parameter to the `getProgressBarJob`).

However, this is very weird. Here especially as there is a clash between dialogBuilder and dialog. dialog is not known when getProgressBarJob is called.

It is probably also weird when dialog titles change with a delay.